src/Abstracts/CollectionResource.php

```diff
@@ -1,83 +1,68 @@
 
+abstract class CollectionResource implements ICollectionResource {
     protected Collection $collection;
     protected Request $request;
+    protected string $table_name;
     protected array $with = [];
     protected array $selectColumns;
-    protected string $table_name;
     protected array $table_columns;
     protected array $columns;
-    protected array $select = ["*"];
+    protected array $select = ['*'];
     protected int $per_page = 50;
-    protected const PAGE_NAME = "page";
-    protected const PER_PAGE_NAME = "per_page";
-    protected const SELECT_COLUMN_NAME = "select";
-    protected const ORDER_COLUMN_NAME = "orderColumn";
-    protected const ORDER_DIRECTION_NAME = "orderDirection";
     protected bool $strict_with_mode = true;
-    protected const PREVENTED_COLUMN_NAMES = [self::ORDER_COLUMN_NAME, self::ORDER_DIRECTION_NAME, self::SELECT_COLUMN_NAME, self::PAGE_NAME, self::PER_PAGE_NAME];
-
-    public function init(Collection $collection, Request $request, $with = []): CollectionResource
-    {
-
+    protected const PAGE_NAME = 'page';
+    protected const PER_PAGE_NAME = 'per_page';
+    protected const SELECT_COLUMN_NAME = 'select';
+    protected const ORDER_COLUMN_NAME = 'orderColumn';
+    protected const ORDER_DIRECTION_NAME = 'orderDirection';
+
+    protected const PREVENTED_COLUMN_NAMES = [
+        self::ORDER_COLUMN_NAME,
+        self::ORDER_DIRECTION_NAME,
+        self::SELECT_COLUMN_NAME,
+        self::PAGE_NAME,
+        self::PER_PAGE_NAME
+    ];
+
+    public function init(Collection $collection, Request $request, array $with = []): CollectionResource {
         $this->collection = $collection;
         $this->request = $request;
         $this->with = $with;
-        $this->per_page = config("datatable.default_per_page", 50);
-        $this->strict_with_mode = config("datatable.strict_with_mode", true);
+        $this->per_page = config('datatable.default_per_page', 50);
+        $this->strict_with_mode = config('datatable.strict_with_mode', true);
+
         $this->build();
 
         return $this;
-
     }
 
-    abstract protected function build(): CollectionResource;
-
-    public function getData(): LengthAwarePaginator
-    {
-
-        return $this->paginate();
-
+    public function getBuilder(): Collection {
+        return $this->collection;
     }
 
-    private function paginate(): LengthAwarePaginator
-    {
-
-        $perPage = (!empty($this->request->{self::PER_PAGE_NAME}) and is_numeric($this->request->{self::PER_PAGE_NAME})) ? $this->request->{self::PER_PAGE_NAME} : $this->per_page;
+    public function getData(): LengthAwarePaginator {
+        $perPage = (!empty($this->request->{self::PER_PAGE_NAME}) && is_numeric($this->request->{self::PER_PAGE_NAME})) ? $this->request->{self::PER_PAGE_NAME} : $this->per_page;
         $page = $this->request->{self::PAGE_NAME};
 
         return new LengthAwarePaginator(
-            $this->collection->forPage($page, $perPage)->values(),
```

Comment:
I guess this changes caused that output of "data" is then for JS object and not array
![image](https://github.com/user-attachments/assets/90a3896d-ed65-4871-b2e4-7457bac176a5)

intead of 
![image](https://github.com/user-attachments/assets/e1efa6d7-4cee-4a6f-b673-30c74622da40)

Reply:
Fixed. Try now please.